substrate/client/transaction-pool/tests/zombienet/limits_30.rs

```diff
@@ -0,0 +1,164 @@
+use std::{path::PathBuf, time::SystemTime};
+
+use zombienet_sdk::{
+	LocalFileSystem, Network as ZNetwork, NetworkConfig, NetworkConfigBuilder, NetworkConfigExt,
+};
+
+use super::{
+	Network, Node, ParachainConfig, RelaychainConfig, DEFAULT_BASE_DIR, DEFAULT_PC_NODE_RPC_PORT,
+	DEFAULT_RC_NODE_RPC_PORT,
+};
+use anyhow::anyhow;
+use which::which;
+
+// A zombienet network with two relaychain 'polkadot' validators and one parachain
+// validator based on yap-westend-live-2022 chain spec.
+pub struct Limits30Network {
+	rc_config: RelaychainConfig,
+	pc_config: ParachainConfig,
+	rc_nodes: Vec<Node>,
+	pc_nodes: Vec<Node>,
+	base_dir: PathBuf,
+}
+
+impl Limits30Network {
+	/// Creates a new [`SmallNetworkLimits30`].
+	pub fn new(
+		rc_config: RelaychainConfig,
+		pc_config: ParachainConfig,
+	) -> Result<Self, anyhow::Error> {
+		// Create temporary network base dir.
+		let datetime: chrono::DateTime<chrono::Local> = SystemTime::now().into();
+		let base_dir =
+			PathBuf::from(format!("{}-{}", DEFAULT_BASE_DIR, datetime.format("%Y%m%d_%H%M%S")));
+		std::fs::create_dir(base_dir.clone()).map_err(|err| anyhow!(format!("{err}")))?;
+		let p_args = vec![
+			"--force-authoring".into(),
+			("--pool-limit", "300").into(),
+			("--pool-kbytes", "2048000").into(),
+			("--rpc-max-connections", "15000").into(),
+			("--rpc-max-response-size", "150").into(),
+			"-lbasic-authorship=info".into(),
+			"-ltxpool=trace".into(),
+			"-lsync=trace".into(),
+			"-laura::cumulus=debug".into(),
+			"-lpeerset=trace".into(),
+			"-lsub-libp2p=debug".into(),
+			"--pool-type=fork-aware".into(),
+			"--state-pruning=1024".into(),
+			"--rpc-max-subscriptions-per-connection=128000".into(),
+		];
+		Ok(Limits30Network {
+			rc_config,
+			pc_config,
+			rc_nodes: vec![
+				Node::new("alice".to_owned(), vec![], true),
+				Node::new("bob".to_owned(), vec![], true),
+			],
+			pc_nodes: vec![
+				Node::new("charlie".to_owned(), p_args.clone(), false),
+				Node::new("dave".to_owned(), p_args.clone(), true),
+				Node::new("eve".to_owned(), p_args.clone(), true),
+				Node::new("ferdie".to_owned(), p_args, true),
+			],
+			base_dir,
+		})
+	}
+}
+
+#[async_trait::async_trait]
+impl Network for Limits30Network {
```

Comment:
Methods seem to be copied into other files. Maybe it does not need to be trait? Or those methods could be provided in the trait. For sure copies should be avoided in the final shape.